analyzers/src/SonarAnalyzer.CSharp/Facade/CSharpSyntaxKindFacade.cs

```diff
@@ -27,6 +27,7 @@ internal sealed class CSharpSyntaxKindFacade : ISyntaxKindFacade<SyntaxKind>
     {
         public SyntaxKind InvocationExpression => SyntaxKind.InvocationExpression;
         public SyntaxKind[] ObjectCreationExpression => new[] {SyntaxKind.ObjectCreationExpression, SyntaxKindEx.ImplicitObjectCreationExpression};
+        public SyntaxKind[] ClassDeclaration => new[] {SyntaxKind.ClassDeclaration, SyntaxKindEx.RecordDeclaration};
```

Comment:
Let's revert this. Member `ClassDeclaration` should return only `SyntaxKind.ClassDeclaration` for those, that would like explicitly subscribe to that single syntax kind.

Let's:
* Use the original approach with abstract method in the rule base, if this is not reusable
* Add new `SyntaxKind[] ClassAndRecordDeclaration` if we can use it at least 3 times directly in existing rules that already contains `ILanguageFacade`.

Reply:
Actually, we don't have to do 1) if we do 2). I've found 2 other rules that benefited from that change